tests/installation/agama.pm

```diff
@@ -0,0 +1,74 @@
+# oSUSE's openQA tests
+#
+# Copyright 2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Installation of Leap or Tumblweed with Agama
+# https://github.com/openSUSE/agama/
+# Maintainer:  Maintainer: Lubos Kocmman <lubos.kocman@suse.com>,
+
+use strict;
+use warnings;
+use testapi;
+use version_utils qw(is_leap is_sle);
+
+
+sub run {
+    my ($self) = @_;
+
+    assert_screen('agama-inst-welcome-product-list');
+
+    if (is_leap('=16')) {
+        assert_and_click('agama-product-leap16');
+        assert_and_click('agama-product-select');
+    } else {    # Default to TW
+        assert_and_click('agama-product-tumbleweed');
+        assert_and_click('agama-product-select');
+    }
+
+    # can take few minutes to get here
+    assert_screen('agama-overview-screen');
+
+    assert_and_click('agama-defining-user');
+    assert_and_click('agama-set-root-password');
+    wait_still_screen 5;
+    # type password,  tab tab, repeat password, tab tab, enter
+    type_password();
+    send_key 'tab';
+    send_key 'tab';
+    wait_still_screen 5;
+    type_password();
+    send_key 'tab';
+    send_key 'tab';
+    send_key 'ret';
```

Comment:
I guess https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/f9b63f416995db83a4d8be70154c10579c899d7d/lib/x11utils.pm#L270 is the proper handler for it. you might want to try out. Not sure it will need any update tho, to integrate some different logic